src/System.Linq/src/System/Linq/Repeat.cs

```diff
@@ -79,10 +79,7 @@ public TResult[] ToArray()
                 TResult[] array = new TResult[_count];
                 if (_current != null)
```

Comment:
But unless there's special recognition of it by the JIT (?), it'll box both value type arguments.  Hence the "efficient" part of my question.